gulpfile.js

```diff
@@ -46,5 +48,25 @@ gulp.task('umd', ['es5', 'es6'], function (cb) {
   })
 });
 
-gulp.task('build', ['clean', 'umd']);
+//formatting
+function doCheckFormat() {
+  return gulp.src(['src/**/*.ts'])
+      .pipe(gulpFormat.checkFormat('file', clangFormat));
+}
+
+gulp.task('check-format', function() {
+  return doCheckFormat().on('warning', function(e) {
+    console.log("NOTE: this will be promoted to an ERROR in the continuous build");
+  });
+});
+
+gulp.task('enforce-format', function() {
+  return doCheckFormat().on('warning', function(e) {
+    console.log("ERROR: You forgot to run clang-format on your change.");
+    console.log("See https://github.com/angular/angular/blob/master/DEVELOPER.md#formatting");
+    process.exit(1);
+  });
+});
+
+gulp.task('build', ['clean', 'umd', 'check-format']);
```

Comment:
the `check-format` task will currently *warn* you but not fail the build (and give you a hint on how to do the fix. the `enforce-format` task should be set up on CI when its ready so builds will fail if they are unformatted. note a couple of the files are already warning :)